### PR TITLE
Remove approve-and-merge

### PR DESCRIPTION
**Pull request by bnchrch**
## Overview
TLDR: We saying goodbye to our dear friend `/approve-and-merge` today.

After a seed was planted by @postamar (https://airbytehq-team.slack.com/archives/C02UF50V9HA/p1696626683363899) and a longer discussion with @git-phu @bgroff @jdpgrailsdev @colesnodgrass @stephane-airbyte and @davinchia its been determined that at this point the `/approve-and-merge` command has started to hurt more than its helped.

**Why?**
No doubt its be a useful tool and a steadfast companion.

However as of late its been responsible for a number of issues
1. The merging of unformatted or linted code
2. Failing Unit Tests on master
3. Failing Integration Tests on master

And the biggest problem by far. **Its gotten us all comfortable with subpar CI**. Unnecessary tests running on docs changes, Flaky unit tests, and in some cases an excessively slow test suite.

Part of the change is to be preventative, but an equal part of removing this command is help us all notice where we can and should improve.

**Why now?**
Over the last 2 months we've
1. Fixed a number of broken and flaky tests
2. Updated Github Actions to allow for required checks to be skipped. (i.e. markdown only changes should not trigger the full test suite)
3. Sped up our entire test suite (Gradle tests, formatting, Connector Acceptance Tests)

And we believe we're now in a much better place where the time to run CI is much more tolerable.

**What do I do now?**
All EM's and many Senior Engineers at Airbyte have admin merge privileges.

If there is an urgent PR ping **any** of them.

If its the middle of the night and youre OC, look at the secondary rotation, they are Admins, page them.

If the integration test is broken, fix it.

and if the test is a known issue and too much effort to fix inside your task ping your EM, get it merged and get the space to fix the underlying issue (or remove it).

**What if I think this is stupid?**
Let us know! We're going to reevaluate this decision inside of the next two weeks.

### CODE REVIEW

**Review by postamar — APPROVED** [review]
Nice! Obviously this will create more work for Connector Ops and Prod Eng, not to mention admin-privilege-havers, still, I agree that it's worth doing now and the connectors CI is sufficiently mature.
